### PR TITLE
feat-dockerize/backend-docs

### PR DESCRIPTION
**Pull request by dasarathg68**
# Description

Addresses issue #20 and adds documentation to get the CNC portal up and running with the frontend, backend and database setup and have also added docker-compose file and options to set it up locally as well. 
The PR fulfils all the criterion mentioned in the issue.

## Type of change

New feature - non breaking change which adds functionality and documentation

### CODE REVIEW

**Review by hermannleboss — CHANGES_REQUESTED** [review]

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -5,9 +5,11 @@
 // Try Prisma Accelerate: https://pris.ly/cli/accelerate-init
 
 generator client {
-  provider = "prisma-client-js"
+  provider      = "prisma-client-js"
+  binaryTargets = ["native"]
```
Comment (hermannleboss) [inline backend/prisma/schema.prisma:9]: Is this necessary

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  frontend:
+    build:
+      context: ./app
+      dockerfile: Dockerfile
+    ports:
+      - "5173:5173" # Adjust port as needed
+    volumes:
+      - ./app:/app
+    command: npm run dev # Command to run for the frontend
```
Comment (hermannleboss) [inline docker-compose.yml:24]: No need to run the comand here

**Review by hermannleboss — APPROVED** [review]

File: backend/prisma/schema.prisma (hunk 1 above)
Comment (dasarathg68) [reply backend/prisma/schema.prisma:9]: It works without it but in the prisma documentation and forums, it is recommended as good practice
https://www.prisma.io/docs/orm/prisma-schema/overview/generators

Comment (hermannleboss) [reply backend/prisma/schema.prisma:9]: okay

**Pull request merged**